### DISCUSSION TITLE
My Feedback After Extensive Testing

### DISCUSSION THREAD

**Discussion by captorxxv** [toeverything/AFFiNE#8466] [General]
I’ve tested a range of similar tools, including Miro, Notion, Obsidian, Appflowy, Craft, Anytype, and a few others. While Affine certainly grabbed my attention as a highly visual individual, there are a few areas where it could improve after spending several days testing both the native macOS app and the self-hosted web app.

	1.	Performance Issues: Regardless of whether I’m using the web app on my server or the native macOS app, I’ve encountered several performance-related bugs and glitches. This results in the app feeling noticeably slower compared to other applications I’ve mentioned.

	2.	Theme and UI Customization: The default themes feel too extreme—too dark in dark mode and too bright in light mode. It would be great to have more customization options to strike a balance that suits different lighting conditions and personal preferences.
	
	3.	Database Functionality: While I understand the team is actively working on a Database 2.0 version, the current iteration leaves room for improvement. For instance, I can only link to another note within the first column. Even when expanding the note, the window opens with an embedded link rather than the actual note, leading to multiple clicks to reach the desired content. Additionally, the note opens in full-screen mode, making it cumbersome to return to the database view.
	
	4.	Task Management: When creating a task within a table and expanding it, I cannot add any subtasks or additional content, which limits the utility of task management features.
	
	5.	Note Templates: The lack of note templates is a significant inconvenience. I find myself manually creating each note, even when they belong to the same category or tag. A template system would streamline this process considerably.
	
	6.	Object Management: While collections and filters within collections are helpful, I would appreciate the ability to assign notes to specific objects. This would enable better categorization and organization within the app.
	
	7.	macOS Native App: The macOS app currently doesn’t allow me to connect to a self-hosted instance, which limits its flexibility for users like myself who prefer self-hosted solutions.
	
	8.	Lack of iPad App: While the iPad supports the web app with Apple Pencil, a native app for iPad would be a valuable addition, particularly for users on the go.
	
	9.	Linking Notes: I would love to see an enhanced note-linking feature, where notes in the database are automatically linked in the “Endless Mode,” improving the overall user experience.
	
	10.	PDF Support: Currently, when dragging and dropping a PDF, only the file is displayed, but not the actual content. Improved PDF handling would enhance usability.
	
	11.	PDF Presentation Mode: In presentation mode, only one static PDF page is displayed, with no ability to navigate through the PDF while presenting. I suggest looking at how Miro implements this feature for inspiration.
	
	12.	PowerPoint Support: The lack of support for running PowerPoint presentations is a limitation. It would be beneficial to integrate PowerPoint file support directly into the presentation mode.

These are just a few of my initial thoughts after testing Affine. I hope this feedback is helpful and provides some insight for future development. The idea behind Affine is fantastic, and the application has immense potential. However, to truly stand out in such a crowded space, it’s important to focus on these areas for improvement.

Great job so far, and I wish the team all the best moving forward!

**Comment by perdakovich** [comment]
did you use ai to write this?
